jenkins/tests/test_upgrade_python_requirements.py

```diff
@@ -78,6 +78,43 @@ def test_changes(self, delete_branch_mock, get_user_mock, create_branch_mock, cr
         assert create_pr_mock.called
         assert not delete_branch_mock.called
 
+    @patch('jenkins.pull_request_creator.PullRequestCreator._get_user',
+           return_value=Mock(name="fake name", login="fake login"))
+    @patch('jenkins.pull_request_creator.PullRequestCreator.github_helper.get_github_instance',
+           return_value=Mock())
+    @patch('jenkins.pull_request_creator.PullRequestCreator.github_helper.repo_from_remote', return_value=Mock())
+    @patch('jenkins.pull_request_creator.PullRequestCreator.github_helper.get_modified_files_list',
+           return_value=["requirements/edx/base.txt", "requirements/edx/coverage.txt"])
+    @patch('jenkins.pull_request_creator.PullRequestCreator.github_helper.get_current_commit', return_value='1234567')
+    # all above this unused params, no need to interact with those mocks
+    @patch('jenkins.pull_request_creator.PullRequestCreator.github_helper.branch_exists', return_value=False)
+    @patch('jenkins.pull_request_creator.PullRequestCreator.github_helper.update_list_of_files', return_value=None)
+    @patch('jenkins.pull_request_creator.PullRequestCreator.github_helper.create_pull_request')
+    @patch('jenkins.pull_request_creator.PullRequestCreator.github_helper.create_branch', return_value=None)
+    @patch('builtins.print')
+    def test_outputs_url_on_success(self, print_mock, create_branch_mock, create_pr_mock,
+                                    update_files_mock, branch_exists_mock, *args):
+        """
+        Ensure that a successful run outputs the URL consumable by github actions
+        """
+        pull_request_creator = PullRequestCreator('--repo_root=../../edx-platform', 'upgrade-branch', [],
+                                                  [], 'Upgrade python requirements', 'Update python requirements',
+                                                  'make upgrade PR', output_pr_url_for_github_action=True)
+        pull_request_creator.create(False)
+
+        assert branch_exists_mock.called
+        assert create_branch_mock.called
+        self.assertEqual(create_branch_mock.call_count, 1)
+        assert update_files_mock.called
+        self.assertEqual(update_files_mock.call_count, 1)
+        assert create_pr_mock.called
+        assert print_mock.called_once
+        found_matching_call = False
+        for call in print_mock.call_args_list:
+            if 'set-output' in call.args[0]:
```

Comment:
Why not check for both 'set-output' and 'generated-pr' in 'call.args[0]'? If any other output is added in future, it may return false positives with current approach.